### DISCUSSION TITLE
ffmpeg

### DISCUSSION THREAD

**Discussion by roots-3d** [AbdBarho/stable-diffusion-webui-docker#273] [General]
Good Morning! First of all many thanks to the devs in the Open Source community for providing free tools distributed to people all over the world.

I wonder how complicated this moment is for programmers when an army of VISUAL ARTISTS are invading Github with the most basic questions, hahaha.

Stable Diffusion Automatic 1111 - Dockerized has been very useful for me (especially for making animations with the Deforum extension), now comes my question.

I'm not finding a way for the container to find the FFMPEG that is working correctly on my Windows (I can call it from the command line). Is there a way for me to configure and find the location of this precious program, thus avoiding me taking my PNGs to Blender just to build the stopmotion?

Because it is an executable of such a small size, do you think you can include it INSIDE the container so that this function is available to everyone without any effort?

Thank you!

**Comment by AbdBarho** [comment]
@Barbolux you can create a script file at `data/config/auto/startup.sh` for installing ffmpeg:
```sh
set -e
apt update
apt install ffmpeg -y
```
And restart your container.
[More info](https://github.com/AbdBarho/stable-diffusion-webui-docker/wiki/Usage#extensions)

I am not sure if this is enough since I don't know your entire workflow... Who / what requires ffmpeg?

**Comment by roots-3d** [comment]
Hello AbdBarho. Happy New Year!

I use Docker Desktop on Windows and FFMPEG is useful in the following operation: within the Deforum extension (which has the main purpose of creating animations), in the "Video Output" tab we have the output_format parameter which can be "PIL gif" (standard, works right out of the box) or "FFMPEG mp4" (does'nt work out of the box).

I'm really asking myself if this would be a desirable function for most users, because for me to have this executable on my Windows it only takes up 80MB. But when installing with apt in the container, it brought almost 500MB of stuff together.

Disk space is not free and could be a nuisance for people who don't even use the Deforum extension.

**Comment by roots-3d** [comment]
The script you suggested didn't work. I don't understand if I did something wrong. However, inspired by your answer, I went looking for solutions myself and I simply managed to open the container's command terminal, in the Windows Docker Desktop GUI, where I inserted the commands apt update & apt install.

Thanks!
